lib/api/apiUtils/authorization/permissionChecks.js

```diff
@@ -507,6 +518,28 @@ function checkIp(value) {
         // notation (e.g., xxx.xxx.xxx.xxx/xx for IPv4). Otherwise,
         // we would accept different ip formats, which is not
         // standard in an AWS use case.
+        // discard to fix tests ?
+
+        /*
+        As seen in https://scality.atlassian.net/browse/ARSN-477
+        the `isValid` does the same parsing as `parseCIDR` but
+        encapsulate in try catch to return a boolean.
+        The `isValid` in catch is useless and the double try
+        discards completely any error from ipaddr.parseCIDR.
+
+        Meaning in case of invalid ip there will be 2 ipaddr.js parsing,
+        reducing perf for an ignored error.
+
+        Fixing this breaks the tests.
+
+        I'm not even sure of the purpose of using ipaddr.js
+        and using our regex after as our regex seems to be
+        more restrictive.
```

Comment:
From what I saw in https://github.com/scality/Arsenal/issues/133, the rationale was to support the ipv6 format used by node, convert it to ipv4 for further usage. The thing is, do we really want to restrict to ipv4 usage? E.g., in node 17, they started favor the use of ipv6 over ipv4. If we get IPv6 addresses as input, this may not work as expected...

Reply:
This convertion is done in another place, here the `checkIp` function validates the format of the IP in the Bucket policy condition